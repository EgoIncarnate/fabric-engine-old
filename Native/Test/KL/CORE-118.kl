function processDataIN( in Size myDataSize, in Data myData )
{
	report( "Size: " + myDataSize + " - Data: " + myData );
}

function processDataIO( in Size myDataSize, io Data myData )
{
	report( "Size: " + myDataSize + " - Data: " + myData );
}

function entry()
{
  var Integer foo;
  processDataIN( foo.dataSize, foo.data );

  var Data fooData = foo.data;
  processDataIO( foo.dataSize, fooData );
  
  var Data null;
  var Size zero = 0;
  
  processDataIN( zero, null );
  processDataIO( zero, null );
}
