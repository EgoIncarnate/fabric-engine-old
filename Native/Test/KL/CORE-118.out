GlobalList
  Function( 'processDataIN' ('processDataIN'), void )
    ParamList
      Param( 'myDataSize', Size, r-value )
      Param( 'myData', Data, r-value )
    CompoundStatement
      StatementList
        Report
          BinOp( ADD )
            BinOp( ADD )
              BinOp( ADD )
                ConstString( 'Size: ' )
                Var('myDataSize')
              ConstString( ' - Data: ' )
            Var('myData')
  Function( 'processDataIO' ('processDataIO'), void )
    ParamList
      Param( 'myDataSize', Size, r-value )
      Param( 'myData', Data, l-value )
    CompoundStatement
      StatementList
        Report
          BinOp( ADD )
            BinOp( ADD )
              BinOp( ADD )
                ConstString( 'Size: ' )
                Var('myDataSize')
              ConstString( ' - Data: ' )
            Var('myData')
  Function( 'entry' ('entry'), void )
    ParamList
    CompoundStatement
      StatementList
        StatementList
          VarDecl( 'foo', Integer )
        ExprStatement
          Call('processDataIN')
            ArgList
              StructMemberOp( 'dataSize' )
                Var('foo')
              StructMemberOp( 'data' )
                Var('foo')
        StatementList
          AssignedVarDecl( VarDecl( 'fooData', Data ), StructMemberOp( 'data' ) )
            StructMemberOp( 'data' )
              Var('foo')
        ExprStatement
          Call('processDataIO')
            ArgList
              StructMemberOp( 'dataSize' )
                Var('foo')
              Var('fooData')
        StatementList
          VarDecl( 'null', Data )
        StatementList
          AssignedVarDecl( VarDecl( 'zero', Size ), ConstSize( 0 ) )
            ConstSize( 0 )
        ExprStatement
          Call('processDataIN')
            ArgList
              Var('zero')
              Var('null')
        ExprStatement
          Call('processDataIO')
            ArgList
              Var('zero')
              Var('null')
Size: 4 - Data: <Opaque>
Size: 4 - Data: <Opaque>
Size: 0 - Data: <Opaque>
Size: 0 - Data: <Opaque>
